Tests/UpstreamIntegrationTests/UpstreamIntegrationTests.swift

```diff
@@ -569,6 +569,267 @@ class UpstreamIntegrationTests: TestBase {
     }
 
+    // MARK: datastream config overrides test
+
+    // Test configOverrides with valid data
+    func testSendEvent_withValidConfigOverrides_receivesExpectedNetworkResponse() {
+        // Setup
+        let interactNetworkRequest = NetworkRequest(urlString: createURLWith(locationHint: edgeLocationHint), httpMethod: .post)!
+        networkService.setExpectationForNetworkRequest(networkRequest: interactNetworkRequest, expectedCount: 1)
+
+        let configOverrides = ["com_adobe_experience_platform": [
+                                    "datasets": [
+                                        "event": [
+                                            "datasetId": "6515e1dbfeb3b128d19bb1e4"
+                                        ]
+
+                                    ]
+                                ],
+                                "com_adobe_analytics": [
+                                    "reportSuites": [
+                                        "mobile5.e2e.rsid2"
+                                    ]
+                                ]]
+        let experienceEvent = ExperienceEvent(xdm: ["xdmtest": "data"], data: ["data": ["test": "data"]], datastreamConfigOverride: configOverrides)
+
+        // Test
+        Edge.sendEvent(experienceEvent: experienceEvent)
+
+        // Verify
+        // Network response assertions
+        let matchingResponses = networkService.getResponsesFor(networkRequest: interactNetworkRequest, timeout: 5)
+
+        XCTAssertEqual(1, matchingResponses.count)
+        XCTAssertEqual(200, matchingResponses.first?.responseCode)
+    }
+
+    // Test configOverrides with dummy data
+    func testSendEvent_withInvalidConfigOverrides_receivesExpectedNetworkResponseError() {
+        // Setup
+        let interactNetworkRequest = NetworkRequest(urlString: createURLWith(locationHint: edgeLocationHint), httpMethod: .post)!
+
+        networkService.setExpectationForNetworkRequest(networkRequest: interactNetworkRequest, expectedCount: 1)
+
+        let expectedErrorJSON = #"""
+        {
+            "status": 400,
+            "title": "Invalid request",
+            "type": "https://ns.adobe.com/aep/errors/EXEG-0113-400"
+        }
+        """#
+
+        let configOverrides = ["test": ["key": "value"]]
+
+        let experienceEvent = ExperienceEvent(xdm: ["xdmtest": "data"], data: ["data": ["test": "data"]], datastreamConfigOverride: configOverrides)
+
+        // Test
+        Edge.sendEvent(experienceEvent: experienceEvent)
+
+        // Verify
+        // Network response assertions
+        let matchingResponses = networkService.getResponsesFor(networkRequest: interactNetworkRequest, timeout: 5)
+
+        XCTAssertEqual(1, matchingResponses.count)
+        assertTypeMatch(expected: getAnyCodable(expectedErrorJSON)!,
+                        actual: getAnyCodable(matchingResponses.first?.responseString ?? ""),
+                        exactMatchPaths: ["status", "title", "type"])
```

Comment:
1. Looks like you are using exact match for all keys, you can simplify this to:
assertExactMatch(expected:..., actual: ...)

or you can include the other keys you may expect in the expected response and use type exact match just for status/title/type, and type match for all the others (see testSendEvent_withInvalidDatastreamID_receivesExpectedError as an example).
Same applies for the other tests below.

2. Is it possible to get more descriptive title/detail for these errors?

Reply:
Hey @emdobrin, I am not getting errors yet m so I don't know what different errors we might see. I will update this once we get the error responses as expected.